File: src/main/java/engine/core/util/color/ColorUtil.java
package engine.core.util.color;

import org.jetbrains.annotations.NotNull;

import java.awt.*;

public class ColorUtil {

    public static float[] getBufferForColor(int verticesCount, @NotNull Color color) {

        int colorLineLength = 3;

        float[] out = new float[verticesCount * colorLineLength];

        float[] colorLine = getColorValues(color);
        for (int i = 0; i < out.length; i += colorLineLength) {
            out[i] = colorLine[0];
            out[i + 1] = colorLine[1];
            out[i + 2] = colorLine[2];
            System.out.println(out[i] + ", " + out[i + 1] + ", " + out[i + 2]);
        }
        return out;
    }

    private static float[] getColorValues(@NotNull Color color) {

        return new float[]{
                color.getRed(),
                color.getGreen(),
                color.getBlue()
        };
    }
}

File: src/main/java/engine/util/color/ColorUtil.kt
package engine.util.color

import java.awt.*

object ColorUtil {
    fun getBufferForColor(verticesCount: Int, color: Color): FloatArray {
        val colorLineLength = 3
        val out = FloatArray(verticesCount * colorLineLength)

        val colorLine = getColorValues(color)
        var i = 0
        while (i < out.size) {
            out[i] = colorLine[0]
            out[i + 1] = colorLine[1]
            out[i + 2] = colorLine[2]
            println(out[i].toString() + ", " + out[i + 1] + ", " + out[i + 2])
            i += colorLineLength
        }
        return out
    }

    private fun getColorValues(color: Color) =
            floatArrayOf(color.red.toFloat(), color.green.toFloat(), color.blue.toFloat())
}
